### ISSUE TITLE
 I would like to ask you how to accurately predict the character order of a word.

### ISSUE THREAD

**Issue by Meicsu199345** [summerlvsong/Aggregation-Cross-Entropy#8]
I recreated your project and found that the input GT was converted into a word list, which had lost its order, and your prediction only provided the number of characters. Only through the two-dimensional matrix position of the network output can barely judge the order, I would like to ask you how to accurately predict the character order of a word.

**Comment by lamhoangtung** [comment]
+1 upvote

**Comment by summerlvsong** [comment]
As mentioned in the paper, to decode the 2D prediction, we flattened the 2D prediction by concatenating each column in order from left to right and top to bottom and then decoded the flattened 1D prediction following the general procedure.

**Comment by chenjun2hao** [comment]
@summerlvsong ,the target label don't need a fixed order?

**Comment by chenjun2hao** [comment]
and i am confused about the 2D example, the label texts don't have a fixed order. if so, how to solve the 1D problem. waiting your reply.

**Comment by summerlvsong** [comment]
During training, we don't need a fixed order for supervision. 
When testing,  for the 2D scene text recogntion problem, we use the hypothesis that character distribute form left to right in the 2D output. Therefore, we can decode the 2D prediction by flattening the 2D prediction by concatenating each column in order from left to right and top to bottom and then decoding the flattened 1D prediction following the general procedure.

**Comment by yinghuozijin** [comment]
@summerlvsong  Thanks for your reply.
